src/WireMockInspector/Views/CodeBlockViewer.cs

```diff
@@ -1,141 +1,114 @@
-    }
+        static CodeBlockViewer()
+        {
+            CodeProperty.Changed.Subscribe(OnCodeChanged);
+        }
 
-    private void OnCodeChanged(MarkdownCode newValue, MarkdownCode NewValue)
-    {
-        SetMarkdown(NewValue);
-    }
+        private static void OnCodeChanged(AvaloniaPropertyChangedEventArgs e)
+        {
+            (e.Sender as CodeBlockViewer)?.OnCodeChanged((ViewModels.MarkdownCode)e.OldValue,
+                (ViewModels.MarkdownCode)e.NewValue);
+        }
 
+        private void OnCodeChanged(MarkdownCode newValue, MarkdownCode NewValue)
+        {
+            SetMarkdown(NewValue);
+        }
 
-    private  void SetMarkdown(ViewModels.MarkdownCode md)
-    {
-        if (md is not null)
+        private void SetMarkdown(ViewModels.MarkdownCode md)
         {
-            if (this.Document is not null)
-            {
-                this.Document.Text = md.rawValue;
-            }
-            else
+            if (md is not null)
             {
-                this.Document = new TextDocument(md.rawValue);
-            }
-            if (_currentLang != md.lang)
-            {
-                _currentLang = md.lang;
-                CodeBlockViewer _textEditor = this;
-
-                if (_registryOptions.GetLanguageByExtension("." + md.lang) is { } languageByExtension)
+                if (this.Document is not null)
                 {
-                    _textMateInstallation.SetGrammar(_registryOptions.GetScopeByLanguageId(languageByExtension.Id));
-                   
-
+                    this.Document.Text = md.rawValue;
+                }
+                else
+                {
+                    this.Document = new TextDocument(md.rawValue);
                 }
 
-
-                if (this.TextArea.TextView.LineTransformers.OfType<DiffLineBackgroundRenderer>().FirstOrDefault() is { } existing)
+                if (_currentLang != md.lang)
                 {
-                    this.TextArea.TextView.LineTransformers.Remove(existing);
+                    _currentLang = md.lang;
+                    CodeBlockViewer _textEditor = this;
+
+                    if (_registryOptions.GetLanguageByExtension("." + md.lang) is { } languageByExtension)
+                    {
+                        _textMateInstallation.SetGrammar(_registryOptions.GetScopeByLanguageId(languageByExtension.Id));
+                    }
+
+                    if (this.TextArea.TextView.LineTransformers.OfType<DiffLineBackgroundRenderer>().FirstOrDefault() is
+                        { } existing)
+                    {
+                        this.TextArea.TextView.LineTransformers.Remove(existing);
+                    }
+
+                    this.TextArea.TextView.LineTransformers.Add(new DiffLineBackgroundRenderer(md.oldTextLines));
+                    if (md.lang == "csharp")
+                    {
+                        this.TextArea.TextView.LineTransformers.Add(new CSharpSyntaxHighlighter());
```

Comment:
Syntax highliting should be handled automatically by Avalonia.Edit and I'm pretty sure it was working correctly so far (for json and csharp). Why did you implemented custom syntax highlighers?

Reply:
Actually it was just all white, i didnt saw any syntax highliting.